reference/helpers.go

```diff
@@ -0,0 +1,34 @@
+package reference
+
+// IsNameOnly returns true if reference only contains a repo name.
+func IsNameOnly(ref Named) bool {
+	if _, ok := ref.(NamedTagged); ok {
+		return false
+	}
+	if _, ok := ref.(Canonical); ok {
+		return false
+	}
+	return true
+}
+
+// SplitName is a helper function which parses the
+// input as a name and returns name plus tag or digest
+func SplitName(s string) (string, string, error) {
+	ref, err := ParseNamed(s)
+	if err != nil {
+		return "", "", err
+	}
+	return ref.Name(), TagOrDigest(ref), nil
+}
+
+// TagOrDigest returns the tag or digest value for the named
+// reference or default tag if is none
+func TagOrDigest(ref Named) string {
```

Comment:
The equivalent logic could be

```
if d, ok := ref.(reference.Digested); ok {
	return d.Digest().String()
}
return reference.WithDefaultTag(ref).Tag()
```

which does not make it useful but still may end up in a helper function by caller. Got from same location as previous block.